src/tests/interpolation/test_interpolation_structured2D.cc

```diff
@@ -277,7 +275,18 @@ CASE( "test_interpolation_structured using fs API multiple levels" ) {
     }
 }
 
-CASE( "test_interpolation_structured using fs API for fieldset" ) {
+template <typename Value>
+struct AdjointTolerance {
+    static const Value value;
+};
+template <>
+const double AdjointTolerance<double>::value = 1.e-8;
+template <>
+const float AdjointTolerance<float>::value = 3.e-1;
+
```

Comment:
Adjoint tolerances should be somewhere near machine epsilon. What are these tolerances relative to? They're only suitable if your test fields have values around 10^4 -- 10^6.

Could you comment further or add stricter tolerances?

Reply:
I think the tolerance should be somewhere around `epsilon * sqrt(num of additions) * (magnitude of field values)`